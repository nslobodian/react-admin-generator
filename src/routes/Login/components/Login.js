import React from 'react'
import PropTypes from 'prop-types'
import s from './Login.scss'
import Field from 'redux-form/es/Field'
import reduxForm from 'redux-form/es/reduxForm'
import FaEnvelope from 'react-icons/lib/fa/envelope'
import FaCertificate from 'react-icons/lib/fa/certificate'

export const LoginForm = ({ handleSubmit, valid, submitFailed, submitting }) =>
  <div className={s.Login}>
    <div className={s.formWrapper}>
      <h4>Herlandia Admin</h4>
      <form
        onSubmit={handleSubmit}
      >
        {submitFailed &&
        <div className='alert alert-danger' role='alert'>Incorrect <b>Email</b> or <b>Password</b></div>
        }
        <div className='form-group'>
          <div className='input-group'>
            <span className='input-group-addon'><FaEnvelope /></span>
            <Field
              component='input'
              className='form-control'
              name='email'
              type='email'
              placeholder='Email'
            />
          </div>
        </div>
        <div className='form-group'>
          <div className='input-group'>
            <span className='input-group-addon'><FaCertificate /></span>
            <Field
              className='form-control'
              component='input'
              name='password'
              type='password'
              placeholder='Password'
            />
          </div>
        </div>
        <button
          type='submit'
          disabled={!valid || submitting}
          className='btn btn-success btn-block btn-lg'
        >
          Login
        </button>
      </form>
    </div>
  </div>

const requiredFields = [
  'email',
  'password',
]

const validate = values => {
  let errors = {}

  requiredFields.forEach(field => {
    if (!values[field]) {
      errors[field] = 'is required'
    }
  })

  return errors
}

LoginForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  valid: PropTypes.bool,
}

export default reduxForm({
  form: 'LoginForm',
  validate,
})(LoginForm)
