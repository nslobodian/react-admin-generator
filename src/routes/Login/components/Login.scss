@import '~styles/shared';

.Login {
  height: 100vh;
  padding: 20%;
  margin: 0 -15px;
  background-color: $jumbotron-bg;

  .formWrapper {
    margin: 0 auto;
    width: 300px;
    background: #fff;
    padding: 20px;
    text-align: center;
  }
}