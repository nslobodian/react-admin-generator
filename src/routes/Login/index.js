import urls from 'utils/urls'

export default (store) => ({
  path: urls.login,
  getComponent (nextState, cb) {
    Promise.all([
      import('./containers/LoginContainer'),
    ]).then((modules) => {
      const Login = modules[0].default

      /*  Return getComponent   */
      cb(null, Login)
    })
  },
})
