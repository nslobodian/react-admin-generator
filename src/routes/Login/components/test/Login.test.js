import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { LoginForm } from '../Login'

describe('(Routes) Login', () => {
  it('Should render LoginForm properly', () => {
  const props = {
    handleSubmit: jest.fn(),
    valid: true,
    submitFailed: true,
    submitting: jest.fn(),
  }
    const wrapper = shallow(
      <LoginForm
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
