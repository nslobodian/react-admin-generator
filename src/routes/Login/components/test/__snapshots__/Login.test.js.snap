// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Routes) Login Should render LoginForm properly 1`] = `
<div
  className="Login"
>
  <div
    className="formWrapper"
  >
    <h4>
      Herlandia Admin
    </h4>
    <form
      onSubmit={[Function]}
    >
      <div
        className="alert alert-danger"
        role="alert"
      >
        Incorrect 
        <b>
          Email
        </b>
         or 
        <b>
          Password
        </b>
      </div>
      <div
        className="form-group"
      >
        <div
          className="input-group"
        >
          <span
            className="input-group-addon"
          >
            <FaEnvelope />
          </span>
          <Field
            className="form-control"
            component="input"
            name="email"
            placeholder="Email"
            type="email"
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <div
          className="input-group"
        >
          <span
            className="input-group-addon"
          >
            <FaCertificate />
          </span>
          <Field
            className="form-control"
            component="input"
            name="password"
            placeholder="Password"
            type="password"
          />
        </div>
      </div>
      <button
        className="btn btn-success btn-block btn-lg"
        disabled={[Function]}
        type="submit"
      >
        Login
      </button>
    </form>
  </div>
</div>
`;
